### PR TITLE
add new fields to pipeline item model

### PR DESCRIPTION
**Pull request by sekharpanja**
### Description of change

Several new fields are added to `company_list_pipelineitem` table as part of next iteration of Pipelines:

      - `"contact_id" uuid NULL`
      - `"sector_id" uuid NULL`
      - `"potential_value" biginteger NULL`
      - `"likelihood_to_win" int NULL`
      - `"expected_win_date" timestamp with time zone NULL`

### Checklist

* [x] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [x] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: changelog/adviser/pipeline_item_fields.db.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Several new fields are added to `company_list_pipelineitem` table as part of next iteration of Pipelines:
+
+      - `"contact_id" uuid NULL`
```
Comment (reupen) [inline changelog/adviser/pipeline_item_fields.db.md:3]: For info, this is rendering as a code block (due to the indentation) which looks like it wasn't intentional.

File: datahub/user/company_list/models.py [hunk 2]
```diff
@@ -95,6 +101,22 @@ class Status(models.TextChoices):
...
+    sector = models.ForeignKey(
+        metadata_models.Sector, blank=True, null=True,
+        on_delete=models.SET_NULL,
+    )
+    potential_value = models.BigIntegerField(blank=True, null=True)
+    likelihood_to_win = models.IntegerField(
+        blank=True, null=True,
+        choices=LikelihoodToWin.choices,
+    )
+    expected_win_date = models.DateTimeField(blank=True, null=True)
```
Comment (reupen) [inline datahub/user/company_list/models.py:119]: Should this be a date or a datetime?

File: datahub/user/company_list/test/factories.py [hunk 3]
```diff
@@ -31,6 +35,11 @@ class PipelineItemFactory(factory.django.DjangoModelFactory):
     company = factory.SubFactory(CompanyFactory)
     adviser = factory.SubFactory(AdviserFactory)
     status = PipelineItem.Status.LEADS
+    contact = factory.SubFactory(ContactFactory, company=factory.SelfAttribute('..company'))
+    sector = factory.SubFactory(SectorFactory)
+    potential_value = 1000000
+    likelihood_to_win = PipelineItem.LikelihoodToWin.MEDIUM
+    expected_win_date = datetime.now() + timedelta(days=180)
```
Comment (reupen) [inline datahub/user/company_list/test/factories.py:42]: Would suggest using `factory.Faker` for this: https://faker.readthedocs.io/en/master/providers/faker.providers.date_time.html

File: datahub/user/company_list/models.py [hunk 4]
```diff
@@ -95,6 +101,22 @@ class Status(models.TextChoices):
...
+        'company.Contact',
+        blank=True, null=True,
+        on_delete=models.CASCADE,
+        related_name='pipeline_items',
+    )
+    sector = models.ForeignKey(
+        metadata_models.Sector, blank=True, null=True,
+        on_delete=models.SET_NULL,
+    )
+    potential_value = models.BigIntegerField(blank=True, null=True)
```
Comment (elcct) [inline datahub/user/company_list/models.py:114]: If it is going to be a money value, shouldn't we use `models.DecimalField`? Would be good to add some `help_text` so there is some context what is being stored.

File: datahub/user/company_list/models.py [hunk 5]
```diff
@@ -85,15 +91,43 @@ class Status(models.TextChoices):
...
+        help_text='Preferred company contact',
+    )
+    sector = models.ForeignKey(
+        metadata_models.Sector, blank=True, null=True,
+        on_delete=models.SET_NULL,
+        help_text='Sector in which the proposal is being worked on',
+    )
+    potential_value = models.DecimalField(
+        blank=True, null=True, max_digits=19, decimal_places=0,
+        help_text='Potential value of the proposal',
```
Comment (reupen) [inline datahub/user/company_list/models.py:121]: Might be nice to add e.g. ` (in £)` to the help text (I assume it's in pounds).

File: datahub/user/company_list/test/factories.py [hunk 6]
```diff
@@ -31,6 +35,11 @@ class PipelineItemFactory(factory.django.DjangoModelFactory):
     company = factory.SubFactory(CompanyFactory)
     adviser = factory.SubFactory(AdviserFactory)
     status = PipelineItem.Status.LEADS
+    contact = factory.SubFactory(ContactFactory, company=factory.SelfAttribute('..company'))
+    sector = factory.SubFactory(SectorFactory)
+    potential_value = 1000000
+    likelihood_to_win = PipelineItem.LikelihoodToWin.MEDIUM
+    expected_win_date = fake.date_between(start_date='today', end_date='+3y')
```
Comment (reupen) [inline datahub/user/company_list/test/factories.py:42]: Would be good to do this via `factory.Faker` e.g. `factory.Faker('future_date', end_date='+3y')`

**Pull request merged**